src/vos/vos_gc.c

```diff
@@ -153,18 +153,26 @@ gc_drain_key(struct vos_gc *gc, struct vos_pool *pool, daos_handle_t coh,
 	struct vos_krec_df *key = umem_off2ptr(&pool->vp_umm, item->it_addr);
 	int		    creds = *credits;
 	int		    rc;
-
+	D_DEBUG(DB_IO, "here");
+	/* flat key */
+	if ((key->kr_bmap & KREC_BF_FLAT) && (gc->gc_type == GC_DKEY)) {
+		D_DEBUG(DB_IO, "here");
+		*empty = true;
+	 	return 0;
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs